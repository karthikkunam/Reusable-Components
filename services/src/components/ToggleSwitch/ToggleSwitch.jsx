import React, { useState } from 'react';
import PropTypes from 'prop-types';

const ToggleSwitch = props => {
  const [id, setId] = useState(props.defaultId ? props.defaultId : props.toggleOptions[0].id);
  const { toggleOptions, onClick, className } = props;

  const onToggleClick = (e, option) => {
    setId(option.id);
    onClick && onClick(e, option);
  };
  return (
    <div className={`switch round pill ${toggleOptions.length > 2 ? 'pill-md' : 'pill-sm'} ${className}`}>
      {toggleOptions &&
        toggleOptions.length > 0 &&
        toggleOptions.map((option, index) => {
          return (
            <>
              <input
                type="radio"
                className="toggle_option"
                id={option.id}
                name="toggle_option"
                onClick={e => onToggleClick(e, option)}
                checked={option.id === id}
              />
              <label key={index} htmlFor={option.id}>
                <p>{option.label}</p>
              </label>
            </>
          );
        })}
      <div className="switch_slider"></div>
    </div>
  );
};

ToggleSwitch.displayName = 'ToggleSwitch';
ToggleSwitch.defaultProps = {
  onClick: () => {},
  className: '',
};

ToggleSwitch.propTypes = {
  toggleOptions: PropTypes.arrayOf(
    PropTypes.objectOf({
      id: PropTypes.oneOf[('first_toggle_md', 'second_toggle_md', 'third_toggle_md')],
      label: PropTypes.string,
      value: PropTypes.string,
    }),
  ).isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  defaultId: PropTypes.oneOf(
    ['first_toggle_md', 'second_toggle_md', 'third_toggle_md'] || ['first_toggle', 'second_toggle'],
  ),
};

export default ToggleSwitch;
