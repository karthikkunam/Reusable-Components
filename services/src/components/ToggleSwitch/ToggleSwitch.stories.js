import React from 'react';
import { action } from '@storybook/addon-actions';
import ToggleSwitch from '.';

export default {
  title: 'ToggleSwitch',
  component: ToggleSwitch,
};

const toggleOptions2 = [
  { id: 'first_toggle', label: 'Cost ($)', value: 'Cost' },
  { id: 'second_toggle', label: 'Retail ($)', value: 'Retail' },
];
export const TwoToggleSwitch = () => <ToggleSwitch toggleOptions={toggleOptions2} onClick={action('toggled')} />;

const toggleOptions = [
  { id: 'first_toggle_md', label: 'Cost ($)', value: 'Cost' },
  { id: 'second_toggle_md', label: 'Retail ($)', value: 'Retail' },
  { id: 'third_toggle_md', label: 'Retail (%)', value: 'Retail Percentage' },
];
export const ThreeToggleSwitch = () => <ToggleSwitch toggleOptions={toggleOptions} onClick={action('toggled')} />;

export const ToggleSwitchWithDefaultView = () => (
  <ToggleSwitch defaultId="second_toggle_md" toggleOptions={toggleOptions} onClick={action('toggled')} />
);
