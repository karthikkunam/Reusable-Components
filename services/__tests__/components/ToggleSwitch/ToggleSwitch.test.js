import React from 'react';
import { shallow } from 'enzyme';
import ToggleSwitch from '../../../src/components/ToggleSwitch/ToggleSwitch.jsx';

const props = {
  toggleOptions: [
    { id: 'first_toggle_md', label: 'Cost ($)', value: 'Cost' },
    { id: 'second_toggle_md', label: 'Retail ($)', value: 'Retail' },
    { id: 'third_toggle_md', label: 'Retail (%)', value: 'Retail Percentage' },
  ],
  onClick: jest.fn(() => {}),
  className: '',
};
describe('ToggleSwitch Component', () => {
  it('renders 2 buttons toggle correctly', () => {
    const props2 = {
      toggleOptions: [
        { id: 'first_toggle', label: 'Cost ($)', value: 'Cost' },
        { id: 'second_toggle', label: 'Retail ($)', value: 'Retail' },
      ],
    };
    const component = shallow(<ToggleSwitch {...props} {...props2} />);
    expect(component).toMatchSnapshot();
  });
  it('renders secondaryView as defaultView for 2 buttons', () => {
    const props1 = {
      defaultId: 'second_toggle',
    };
    const component = shallow(<ToggleSwitch {...props} {...props1} />);
    expect(component).toMatchSnapshot();
  });

  it('renders 3 buttons toggle correctly', () => {
    const component = shallow(<ToggleSwitch {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders secondaryView as defaultView for 3 buttons', () => {
    const props1 = {
      defaultId: 'second_toggle_md',
    };
    const component = shallow(<ToggleSwitch {...props} {...props1} />);
    expect(component).toMatchSnapshot();
  });
  it('should toggle buttons onClick of ', () => {
    const component = shallow(<ToggleSwitch {...props} />);
    component
      .find('input')
      .at(0)
      .simulate('click', {
        onToggleClick: () => {},
        setId: () => {},
      });
    expect(component).toMatchSnapshot();
  });
});
