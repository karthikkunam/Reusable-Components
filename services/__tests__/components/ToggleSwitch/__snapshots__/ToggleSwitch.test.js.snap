// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Component renders 2 buttons toggle correctly 1`] = `
<div
  className="switch round pill pill-sm mb-4 "
>
  <input
    checked={true}
    className="toggle_option"
    id="first_toggle"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="first_toggle"
    key="0"
  >
    <p>
      Cost ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="second_toggle"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="second_toggle"
    key="1"
  >
    <p>
      Retail ($)
    </p>
  </label>
  <div
    className="switch_slider"
  />
</div>
`;

exports[`ToggleSwitch Component renders 3 buttons toggle correctly 1`] = `
<div
  className="switch round pill pill-md mb-4 "
>
  <input
    checked={true}
    className="toggle_option"
    id="first_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="first_toggle_md"
    key="0"
  >
    <p>
      Cost ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="second_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="second_toggle_md"
    key="1"
  >
    <p>
      Retail ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="third_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="third_toggle_md"
    key="2"
  >
    <p>
      Retail (%)
    </p>
  </label>
  <div
    className="switch_slider"
  />
</div>
`;

exports[`ToggleSwitch Component renders secondaryView as defaultView for 2 buttons 1`] = `
<div
  className="switch round pill pill-md mb-4 "
>
  <input
    checked={false}
    className="toggle_option"
    id="first_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="first_toggle_md"
    key="0"
  >
    <p>
      Cost ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="second_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="second_toggle_md"
    key="1"
  >
    <p>
      Retail ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="third_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="third_toggle_md"
    key="2"
  >
    <p>
      Retail (%)
    </p>
  </label>
  <div
    className="switch_slider"
  />
</div>
`;

exports[`ToggleSwitch Component renders secondaryView as defaultView for 3 buttons 1`] = `
<div
  className="switch round pill pill-md mb-4 "
>
  <input
    checked={false}
    className="toggle_option"
    id="first_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="first_toggle_md"
    key="0"
  >
    <p>
      Cost ($)
    </p>
  </label>
  <input
    checked={true}
    className="toggle_option"
    id="second_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="second_toggle_md"
    key="1"
  >
    <p>
      Retail ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="third_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="third_toggle_md"
    key="2"
  >
    <p>
      Retail (%)
    </p>
  </label>
  <div
    className="switch_slider"
  />
</div>
`;

exports[`ToggleSwitch Component should toggle buttons onClick of  1`] = `
<div
  className="switch round pill pill-md mb-4 "
>
  <input
    checked={true}
    className="toggle_option"
    id="first_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="first_toggle_md"
    key="0"
  >
    <p>
      Cost ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="second_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="second_toggle_md"
    key="1"
  >
    <p>
      Retail ($)
    </p>
  </label>
  <input
    checked={false}
    className="toggle_option"
    id="third_toggle_md"
    name="toggle_option"
    onClick={[Function]}
    type="radio"
  />
  <label
    htmlFor="third_toggle_md"
    key="2"
  >
    <p>
      Retail (%)
    </p>
  </label>
  <div
    className="switch_slider"
  />
</div>
`;
